### ISSUE TITLE
checkPermission() always returns LocationPermission.deniedForever when App Permission Settings is "Ask every time"

### ISSUE THREAD

**Issue by S-Man42** [Baseflow/flutter-geolocator#532]
Hi,

I tried following with an Android App (API Level 29):

```
  var _currentPermissionGranted = false;

  Future<LocationPermission> _checkLocationPermission() {
    checkPermission().then((permission) {
      print(permission);  // PROBLEM: always returns LocationPermission.deniedForever

      if (permission == null)
        return;

      var newPermission = false;
      if ([LocationPermission.always, LocationPermission.whileInUse].contains(permission))
        newPermission = true;

      if (permission == LocationPermission.denied)
        requestPermission().then((status) {
          _currentPosition = null;  // _currentPosition will be set through PositionStreamSubscription later
          setState(() {});
        });

      if (newPermission != _currentPermissionGranted) {
        setState(() {
          _currentPermissionGranted = newPermission;
        });
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    _checkLocationPermission();
   
    if (_currentPermissionGranted) {doSth();}

    return ...
  }
```

When I start my app for the first time, I get the permission reguest. Then, I choose "Ask every time", then the `checkPermssion()` returns `LocationPermission.whileInUse` and the position data is streamed.

But on every further restart/widget initialisation, `checkPermission()` returns `LocationPermission.deniedForever` for any reasons. However, the System App Settings still shows "Ask every time". Is there a problem with how did I use the API?

**Comment by aramayyes** [comment]
I'm having similar issue on Android Emulator API 30.

**Comment by mvanbeusekom** [comment]
@S-Man42 , @aramayyes , thank you for reporting this issue. I am looking into it.

**Comment by aramayyes** [comment]
@mvanbeusekom any news?

**Comment by mgav** [comment]
I'm a novice, but wiping the emulator's data* might help.

_*Open the AVD manager and in the list of devices, all the way on the right hand side you click the dropdown arrow and select Wipe Data_

**Comment by aramayyes** [comment]
@mgav, thank you for reply, but it didn't help.

**Comment by Galti** [comment]
Any news? I don't want to change this package for this issue. 23 days and this issue still open. It smells like react-native package.

**Comment by mvanbeusekom** [comment]
Hi @aramayyes, @Galti, 

Have been thrown of on this issue since the OP mentioned this issue occurs on API 29 while this issue actually only happens on API 30 (selecting the "One time" permission is introduced in Android 11).

Long story short, I can reproduce the issue on API 30 and will be introducing a fix today (as part of release 6.1.0).

**Comment by averstuyf** [comment]
I've experienced this today on a fully updated Google Pixel XL2 and flutter-geolocator 6.1.5.

To reproduce: Run the example app (I've tried the default target API 28 and 30) on Android 11, API 30. Quit and close the example app. Long press the example app icon > App info > Permissions > Location and choose 'Ask every time'. Open the example app and choose 'getPermissionStatus'. 'LocationPermission.deniedForever' appears.

**Comment by dsgriffin** [comment]
Still happening on API 30 Emulator

**Comment by mvanbeusekom** [comment]
@dsgriffin which version of the Geolocator are you using? This problem has been solved in version 7.0.0 of the Geolocator, you can read detailed explanation on our wiki: https://github.com/Baseflow/flutter-geolocator/wiki/Breaking-changes-in-7.0.0.

**Comment by dsgriffin** [comment]
@mvanbeusekom I'm using 7.0.1, and testing with the `deny` and `ask every time` permissions on Android Emulator SDK 30.

```
      var locationPermission = await Geolocator.checkPermission();

      if (locationPermission == LocationPermission.deniedForever) {
          return;
      }

      if (locationPermission == LocationPermission.denied) {
        await showRequestLocationDialog();

        locationPermission = await Geolocator.requestPermission();

        // User still denied it after our request
        if (locationPermission != LocationPermission.whileInUse && locationPermission != LocationPermission.always) {
          return;
        }
      }

      Position currentPosition = await locationService.getPosition();
```

I'm having two issues; 

1) now `checkPermission` returns `denied` instead of `deniedForever`, I am unsure how to show the custom alert dialog explaining why we need the permission to just users who denied once

2) `await Geolocator.requestPermission` returns `deniedForever` when a user has `Ask Every Time`, both on emulator and on real devices.
